src/main/java/btdex/core/ContractState.java

```diff
@@ -405,10 +405,10 @@ private void buildHistory(Transaction[] txs) {
 					&& tx.getAppendages()[0] instanceof PlaintextMessageAppendix) {
 
 				PlaintextMessageAppendix appendMessage = (PlaintextMessageAppendix) tx.getAppendages()[0];
-				if(tx.getSender().equals(at.getCreator()) && appendMessage.getMessage().startsWith("{")) {
+				String jsonData = appendMessage.getMessage();
```

Comment:
Maybe better name this variable as `messageString` or something like that, see the comment below.

Reply:
I agree. Good point.